### ISSUE TITLE
Proposal: Improve keyboard navigation in text files

### ISSUE THREAD

**Issue by SetTrend** [Microsoft/vscode#3249]
I feel that the current stop points used by Code for navigating through a text document using keyboard combinations <kbd>CTRL</kbd> + <kbd>LEFT ARROW</kbd> and <kbd>CTRL</kbd> + <kbd>RIGHT ARROW</kbd>, resp., are quite confusing.

Apparently, the cursor isn't moved to the next word boundary ([regex anchor `\b`](https://msdn.microsoft.com/en-us/library/h5181w5w%28v=vs.110%29.aspx#WordBoundary)) as is done by almost any editor I know. ... Frankly, I fail trying to understand what the exact algorithm behind the current keyboard navigation behaviour in Code may be.

Can we have Code behave like Visual Studio 2015 when navigating through text files using keyboard combinations <kbd>CTRL</kbd> + <kbd>LEFT ARROW</kbd> and <kbd>CTRL</kbd> + <kbd>RIGHT ARROW</kbd>, resp.?

**Comment by joaomoreno** [comment]
ping @bpasero

**Comment by alexandrudima** [comment]
@SetTrend please check out the upcoming 0.10.10 where we've refurbished the word navigation

**Comment by SetTrend** [comment]
Great work. I'm looking forward to it.

Thanks! :clap:

**Comment by alexandrudima** [comment]
@SetTrend To clarify, the navigation is refurbished, but Ctrl+Right will still go to the end of the words, and not to the beginning of words as in Visual Studio 2015.

We're tracking this in #3577 and considering making it into an editor option.

**Comment by SetTrend** [comment]
Excellent. I'm sure this'll give much comfort to Windows users when switching between applications.

Thanks for keeping up the good work! :sunflower:
